myJoin.pl

```diff
@@ -36,7 +36,12 @@
 "help" => \$help) or die("Error in command line arguments -h for help\n");
 
 #process field argument
-@field = split(",", $field);
+@fields = split(",", $field);
+my %field_names = map {$_ => 1} @fields;
```

Comment:
`%field_names` is added here for rapid lookup of names in your subroutine below (rather than the slower approach of whole-array searches).